workflow/rules/common.smk

```diff
@@ -202,12 +262,33 @@ def all_input(wildcards):
                 "results/plots/ma/{model}.ma-plots.pdf",
                 "results/plots/qq/{model}.qq-plots.pdf",
                 "results/tables/diffexp/{model}.transcripts.diffexp.tsv",
-                # "results/plots/diffexp-heatmap/{model}.diffexp-heatmap.pdf", # see rule plot_diffexp_heatmap
                 "results/tables/logcount-matrix/{model}.logcount-matrix.tsv",
+                "results/sleuth/{model}.samples.tsv",
+                "results/datavzrd-reports/diffexp-{model}",
             ],
             model=config["diffexp"]["models"],
         )
     )
+    if config["diffexp"]["genes_of_interest"]["activate"] == True:
+        wanted_input.extend(
+            expand(
+                [
+                    "results/plots/diffexp-heatmap/{model}.diffexp-heatmap.{mode}.pdf",
+                ],
+                model=config["diffexp"]["models"],
+                mode=["topn", "predefined"],
+            )
+        )
+    else:
+        wanted_input.extend(
+            expand(
+                [
+                    "results/plots/diffexp-heatmap/{model}.diffexp-heatmap.{mode}.pdf",
+                ],
+                model=config["diffexp"]["models"],
+                mode=["topn"],
+            )
+        )
```

Comment:
Redundant, just change what you pass to mode depending on whether genes_of_interest is activated or not.

Reply:
A small doubt, As we have commented the `genes_of_interest` in the config file, I am not clear how can we pass `genes_of_interest` variable by default.

Reply:
You could use .get() here, but actually it is indeed better to not comment out genes_of_interest. Instead, just have activate: false as default for it, as you had before. I withdraw my change request for that.

Reply:
From what I understood based on the comments above, I updated the output based on the activation of `genes_of_interest` parameter.